File: DiKErnel/Test/DiKErnel.Integration.TestUtil/ProfileDataAssertHelper.cpp
// Copyright (C) Stichting Deltares 2022. All rights reserved.
//
// This file is part of DiKErnel.
//
// DiKErnel is free software: you can redistribute it and/or modify
// it under the terms of the GNU Lesser General Public License as published by
// the Free Software Foundation, version 3.
// 
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
// GNU Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public License
// along with this program. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
//
// All names, logos, and references to "Deltares" are registered trademarks of
// Stichting Deltares and remain full property of Stichting Deltares at all times.
// All rights reserved.

#include "ProfileDataAssertHelper.h"

#include <gtest/gtest.h>

namespace DiKErnel::Integration::TestUtil
{
    using namespace Core;

    void ProfileDataAssertHelper::AssertProfilePoint(
        const double expectedX,
        const double expectedZ,
        const ProfilePoint& actualProfilePoint)
    {
        ASSERT_EQ(expectedX, actualProfilePoint.GetX());
        ASSERT_EQ(expectedZ, actualProfilePoint.GetZ());
    }

    void ProfileDataAssertHelper::AssertProfileSegment(
        const double expectedLowerPointX,
        const double expectedLowerPointZ,
        const double expectedUpperPointX,
        const double expectedUpperPointZ,
        const double expectedRoughnessCoefficient,
        const ProfileSegment& actualProfileSegment)
    {
        AssertProfilePoint(expectedLowerPointX, expectedLowerPointZ, actualProfileSegment.GetLowerPoint());
        AssertProfilePoint(expectedUpperPointX, expectedUpperPointZ, actualProfileSegment.GetUpperPoint());
        ASSERT_EQ(expectedRoughnessCoefficient, actualProfileSegment.GetRoughnessCoefficient());
    }

    void ProfileDataAssertHelper::AssertCharacteristicPoint(
        const ProfilePoint& expectedProfilePoint,
        const CharacteristicPointType expectedCharacteristicPointType,
        const CharacteristicPoint& actualCharacteristicPoint)
    {
        ASSERT_EQ(&expectedProfilePoint, &actualCharacteristicPoint.GetProfilePoint());
        ASSERT_EQ(expectedCharacteristicPointType, actualCharacteristicPoint.GetCharacteristicPointType());
    }
}

File: Test/DiKErnel.Integration.TestUtil/ProfileDataAssertHelper.cs
// Copyright (C) Stichting Deltares and State of the Netherlands 2025. All rights reserved.
//
// This file is part of DiKErnel.
//
// This program is free software; you can redistribute it and/or modify it under the terms of
// the GNU Lesser General Public License as published by the Free Software Foundation; either
// version 3 of the License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
// without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
// See the GNU Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public License along with this
// program; if not, see <https://www.gnu.org/licenses/>.
//
// All names, logos, and references to "Deltares" are registered trademarks of Stichting
// Deltares and remain full property of Stichting Deltares at all times. All rights reserved.

using DiKErnel.Core.Data;
using NUnit.Framework;

namespace DiKErnel.Integration.TestUtil
{
    internal static class ProfileDataAssertHelper
    {
        public static void AssertProfileSegment(double startPointX, double startPointZ, double endPointX,
                                                double endPointZ, double roughnessCoefficient,
                                                ProfileSegment profileSegment)
        {
            AssertProfilePoint(startPointX, startPointZ, profileSegment.StartPoint);
            AssertProfilePoint(endPointX, endPointZ, profileSegment.EndPoint);
            Assert.That(profileSegment.RoughnessCoefficient, Is.EqualTo(roughnessCoefficient));
        }

        public static void AssertCharacteristicPoint(ProfilePoint profilePoint,
                                                     CharacteristicPointType characteristicPointType,
                                                     CharacteristicPoint characteristicPoint)
        {
            Assert.That(characteristicPoint.ProfilePoint, Is.SameAs(profilePoint));
            Assert.That(characteristicPoint.CharacteristicPointType, Is.EqualTo(characteristicPointType));
        }

        public static void AssertForeshore(double slope, double bottomZ, Foreshore foreshore)
        {
            Assert.That(foreshore.Slope, Is.EqualTo(slope));
            Assert.That(foreshore.BottomZ, Is.EqualTo(bottomZ));
        }

        private static void AssertProfilePoint(double x, double z, ProfilePoint profilePoint)
        {
            Assert.That(profilePoint.X, Is.EqualTo(x));
            Assert.That(profilePoint.Z, Is.EqualTo(z));
        }
    }
}